debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/pgoutput/PgOutputMessageDecoder.java

```diff
@@ -123,54 +128,86 @@ public PgOutputMessageDecoder(MessageDecoderContext decoderContext, PostgresConn
             MessageType type = MessageType.forType((char) buffer.get());
             LOGGER.trace("Message Type: {}", type);
+
             switch (type) {
                 case TYPE:
                 case ORIGIN:
-                    // TYPE/ORIGIN
-                    // These should be skipped without calling shouldMessageBeSkipped. DBZ-5792
-                    LOGGER.trace("{} messages are always skipped without calling shouldMessageBeSkipped", type);
                     return true;
+
                 case TRUNCATE:
                     if (!isTruncateEventsIncluded()) {
-                        LOGGER.trace("{} messages are being skipped without calling shouldMessageBeSkipped", type);
+                        LOGGER.trace("TRUNCATE message skipped as truncate events are not included.");
                         return true;
                     }
-                    // else delegate to super.shouldMessageBeSkipped
                     break;
-                default:
-                    // call super.shouldMessageBeSkipped for rest of the types
-            }
-            final boolean candidateForSkipping = super.shouldMessageBeSkipped(buffer, lastReceivedLsn, startLsn, walPosition);
-            switch (type) {
+
+                case RELATION:
                 case COMMIT:
+                    return false;
+
                 case BEGIN:
-                case RELATION:
-                    // BEGIN
-                    // These types should always be processed due to the nature that they provide
-                    // the stream with pertinent per-transaction boundary state we will need to
-                    // always cache as we potentially might reprocess the stream from an earlier
-                    // LSN point.
-                    //
-                    // RELATION
-                    // These messages are always sent with a lastReceivedLSN=0; and we need to
-                    // always accept these to keep per-stream table state cached properly.
-                    LOGGER.trace("{} messages are always reprocessed", type);
+                    this.commitLsn = Lsn.valueOf(buffer.getLong());
                     return false;
+
+                case LOGICAL_DECODING_MESSAGE:
+                    boolean isTransactional = buffer.get() == 1;
+                    if (!isTransactional) {
+                        this.commitLsn = Lsn.valueOf(buffer.getLong());
+                    }
+                    break;
+
                 default:
-                    // INSERT/UPDATE/DELETE/TRUNCATE/LOGICAL_DECODING_MESSAGE
-                    // These should be excluded based on the normal behavior, delegating to default method
-                    return candidateForSkipping;
+                    // Unknown or unhandled message types
+                    LOGGER.debug("Unhandled message type: {}", type);
+                    break;
             }
+
+            // Post-processing logic
+            if (lsnSearchCompleted) {
```

Comment:
Can you add a short comment for each `if` condition, describing the state which is handled in the given condition?